activerecord/test/cases/associations/eager_test.rb

```diff
@@ -1487,6 +1487,21 @@ def test_eager_load_multiple_associations_with_references
     assert_equal posts(:welcome), post
   end
 
+  test "preloading of non-relations" do
+    class SpecialPostPreloader < ActiveRecord::Associations::Preloader::HasMany
+      def initialize(authors)
+        super(Post, authors, Author.reflect_on_association(:posts), ActiveRecord::Relation.new(nil, nil, nil))
+      end
+
+      def records_for(author_ids)
+        # to_a on an actual relation is called to simulate a real case
+        Post.where(author_id: author_ids).to_a
+      end
+    end
+
+    SpecialPostPreloader.new(Author.first(10)).run(nil)
```

Comment:
That's true - I'm using an internal API that may break for me at any time, I am aware and have my own tests to cover this. This doesn't mean the internal API shouldn't make sense - I don't see a reason why the `records_for` method should be forced to return an `ActiveRecord::Relation`, it is expected to return records associated to the given records which is an `Enumerable`. So this test might not make sense in the test suite (as it will break for no good reason), but the above change does.